### ISSUE TITLE
Enhancement: [naming-convention] non-strict PascalCase vs UPPER_CASE

### ISSUE THREAD

**Issue by Faithfinder** [typescript-eslint/typescript-eslint#6213]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/naming-convention/

### Description

I know it's intentional right now, and has been declined quite recently (#5486).
However I feel like there's value in allowing `ObjectID` and forbidding `OBJECTID`.
I say, non-strict pascal case should have a number of allowed consecutive uppercase letters. I think there's no place to have settings for a format right now? I'd take an arbitrary 3-4 symbol restriction. Maybe as a `StrictishPascalCase` format, to avoid breaking changes.

### Fail

```typescript
var OBJECTID = '123';
```

### Pass

```typescript
var ObjectID = '123';
```

**Comment by bradzacher** [comment]
There's ultimately no way to get thus right in the general case because acronyms can be chained in valid ways that conflict with whatever scheme you come up with. It's just a fun property of the English language.

Eg you say 4 is the max... Well `ATMID` would be disallowed then even though an engineer can see that it should be clearly valid.

The best way to create a solid and predictable naming convention is to set a rule in your codebase and follow it. That rule should be either acronyms follow strict pascal case, or they don't. Why? Because that way you have a coding style that is consistent and always works the same - making it predictable and easy to understand. Stylistic linting should be predictable and simple so that you get out of the engineer's way so they can focus on solving problems.

If you want to get fancy you could enforce strict pascal case but use the regex filter to enforce certain acronyms are always upper case - but I'd advice against it as it makes things complicated and can false positive accidentally (eg enforce ID vs Id and now you enforce `IDentifier` instead of `Identifier`).

For the above reasoning the request is going to remain rejected. Thanks for filing an issue!
